test/indexer_integration_test.go

```diff
@@ -552,40 +563,50 @@ func thereAreAssetsInTheResponseTheFirstIs(numAssetsInResponse, assetIdFirstAsse
 }
 
 // @indexer.applications
-func iUseToSearchForApplicationsWithAndToken(indexer, limit, appId int, token string) error {
+func deprecatedIUseToSearchForApplicationsWithAndToken(indexer, limit, appId int, token string) error {
+	return iUseToSearchForApplicationsWithAndToken(indexer, limit, appId, "false", token)
+}
+
+func iUseToSearchForApplicationsWithAndToken(indexer, limit, appId int, includeAll, token string) error {
 	ic := indexerClients[indexer]
 	var err error
-	response, err = ic.SearchForApplications().
+	query := ic.SearchForApplications().
 		ApplicationId(uint64(appId)).
 		Limit(uint64(limit)).
-		Next(token).Do(context.Background())
+		Next(token)
+	if ia, err := strconv.ParseBool(includeAll); err == nil && ia {
+		query.IncludeAll(ia)
+	}
+	response, err = query.Do(context.Background())
 	return err
 }
 
 func theParsedResponseShouldEqual(jsonfileName string) error {
-	var err error
 	var responseJson string
 
 	baselinePath := path.Join("./features/resources/", jsonfileName)
 	responseJson = string(json.Encode(response))
 
-	jsonfile, err := os.Open(baselinePath)
-	if err != nil {
-		return err
-	}
-	fileBytes, err := ioutil.ReadAll(jsonfile)
-	if err != nil {
-		return err
-	}
-	_, err = EqualJson(string(fileBytes), responseJson)
-	return err
+	return VerifyResponse(baselinePath, responseJson)
+}
+
+func deprecatedIUseToLookupApplicationWith(indexer, appid int) error {
+	return iUseToLookupApplicationWithAnd(indexer, appid, "false")
 }
 
-func iUseToLookupApplicationWith(indexer, appid int) error {
+func iUseToLookupApplicationWithAnd(indexer, appid int, includeAll string) error {
 	ic := indexerClients[indexer]
 	var err error
-	response, err = ic.LookupApplicationByID(uint64(appid)).Do(context.Background())
-	return err
+	query := ic.LookupApplicationByID(uint64(appid))
+	if ia, err := strconv.ParseBool(includeAll); err == nil && ia {
+		query.IncludeAll(ia)
+	}
+	response, err = query.Do(context.Background())
+	// allow 404...
```

Comment:
Technically all of these functions could be changed to only return an error when `err` is not an HTTP error. For now, this is the only one that needs that functionality.